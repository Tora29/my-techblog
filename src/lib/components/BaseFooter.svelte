<script lang=ts>
  
</script>

<footer class="footer footer-center bg-base-300 text-base-content p-4 absolute bottom-0">
  <aside>
    <p>Copyright Â© {new Date().getFullYear()} - All right reserved by Tora29 Industries Ltd</p>
  </aside>
</footer>
