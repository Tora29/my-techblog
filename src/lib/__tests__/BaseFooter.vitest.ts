import { render } from '@testing-library/svelte';
import { expect, test } from 'vitest';
import '@testing-library/jest-dom';
import BaseFooter from '$lib/components/BaseFooter.svelte';

describe('Footer Component', () => {
	test('フッターをデフォルトプロパティでレンダリング', () => {
		const { getByText, container } = render(BaseFooter);

		// コピーライトテキストが期待通りであることを確認
		const copyrightText = getByText(
			/Copyright © \d{4} - All right reserved by Tora29 Industries Ltd/i
		);
		expect(copyrightText).toBeInTheDocument();

		// デフォルトクラスが適用されているかを確認
		const footerElement = container.querySelector('footer');
		expect(footerElement).toHaveClass('footer-center', 'bg-primary', 'p-4');

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});

	test('フッターをカスタムプロパティでレンダリング', () => {
		const { container } = render(BaseFooter, {
			props: {
				color: 'bg-secondary',
				padding: 'p-8'
			}
		});

		// カスタムクラスが適用されているかを確認
		const footerElement = container.querySelector('footer');
		expect(footerElement).toHaveClass('bg-secondary', 'p-8');

		// スナップショットを取得
		expect(container).toMatchSnapshot();
	});
});
