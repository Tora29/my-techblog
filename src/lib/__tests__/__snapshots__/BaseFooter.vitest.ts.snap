// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer Component > フッターをカスタムプロパティでレンダリング 1`] = `
<div>
  <footer
    class="footer footer-center bg-secondary p-8 svelte-xk28wq"
  >
    <aside>
      <p>
        Copyright © 
        2024
         - All right reserved by Tora29 Industries Ltd
      </p>
    </aside>
  </footer>
</div>
`;

exports[`Footer Component > フッターをデフォルトプロパティでレンダリング 1`] = `
<div>
  <footer
    class="footer footer-center bg-primary p-4 svelte-xk28wq"
  >
    <aside>
      <p>
        Copyright © 
        2024
         - All right reserved by Tora29 Industries Ltd
      </p>
    </aside>
  </footer>
</div>
`;
